lib/matplotlib/ticker.py

```diff
@@ -1685,6 +1685,33 @@ def __init__(self, *args, **kwargs):
         self.set_params(**self.default_params)
         self.set_params(**kwargs)
 
+    @staticmethod
+    def _validate_steps(steps):
+        if not np.iterable(steps):
+            raise ValueError('steps argument must be a sequence of numbers '
+                             'from 1 to 10')
+        steps = np.asarray(steps)
+        if np.any(np.diff(steps) <= 0):
+            raise ValueError('steps argument must be uniformly increasing')
+        if np.any((steps > 10) | (steps < 1)):
+            warnings.warn('Steps argument should be a sequence of numbers\n'
+                          'increasing from 1 to 10, inclusive. Behavior with\n'
+                          'values outside this range is undefined, and will\n'
+                          'raise a ValueError in future versions of mpl.')
+        if steps[0] != 1:
+            steps = np.hstack((1, steps))
+        if steps[-1] != 10:
+            steps = np.hstack((steps, 10))
+        return steps
+
+    @staticmethod
+    def _staircase(steps):
+        # Make an extended staircase within which the needed
+        # step will be found.  This is probably much larger
+        # than necessary.
+        flights = (0.1 * steps[:-1], steps, 10 * steps[1])
```

Comment:
OK, I never noticed this on the previous PR, but why is the third entry ten times the second step, and not ten times the second-to-the-last steps in the same way as the first element?